dom/src/main/scala/fs2/dom/ResizeObserver.scala

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2022 Arman Bilge
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fs2.dom
+
+import scalajs.js
+import org.scalajs.dom
+import cats.effect.kernel.Async
+import cats.effect.std.Dispatcher
+import cats.effect.kernel.Resource
+
+trait ResizeObserver[F[_]] {
+
+  def observe(target: dom.Element): F[Unit]
+
+  def observe(target: dom.Element, options: dom.ResizeObserverOptions): F[Unit]
+
+  def unobserve(target: dom.Element): F[Unit]
```

Comment:
Hmmmm ... in the Scala 3 side, we have wrappers for `dom.Element` actually. I wonder if we should keep this `ResizeObserver` on Scala 2 side, and make a different interface on Scala 3 🤔

Reply:
By the way, why are those wrappers needed for Scala 3?

Reply:
It's so you can safely access the DOM in tagless final style without passing an `Async` or `Sync` constraint everywhere. e.g. look at the latest Calico demos.

https://armanbilge.github.io/calico/demos/hello-world.html